src/sdk/pynni/nni/trial.py

```diff
@@ -147,9 +148,10 @@ def report_final_result(metric):
     metric:
         serializable object.
     """
-    assert _params is not None, 'nni.get_next_parameter() needs to be called before report_final_result'
+    assert _params or trial_env_vars.NNI_PLATFORM is None, \
```

Comment:
Why `_params is not None` is now `_params`?

Reply:
I believe `_params` must not be `{}` either.